sources/Context.cpp

```diff
@@ -51,14 +51,30 @@ Context::Context(const std::string& networkFilepath, const std::string& configFi
     }
     networkManager_.onNode(algo::SlackNodeAlgorithm(slackNode_));
   }
+
+  networkManager_.onNode(algo::ConnexityAlgorithm(mainConnexNodes_));
 }
 
-void
+bool
+Context::checkConnexity() const {
+  // The slack node must be in the main connex component
+  return std::find_if(mainConnexNodes_.begin(), mainConnexNodes_.end(),
+                      [this](const std::shared_ptr<inputs::Node>& node) { return node->id == slackNode_->id; }) != mainConnexNodes_.end();
+}
+
+bool
 Context::process() {
   // Process all algorithms on nodes
   networkManager_.walkNodes();
 
   LOG(info) << MESS(SlackNode, slackNode_->id, static_cast<unsigned int>(slackNodeOrigin_)) << LOG_ENDL;
+
+  if (!checkConnexity()) {
+    LOG(error) << MESS(ConnexityError, slackNode_->id) << LOG_ENDL;
```

Comment:
If the selection of the slack node is automatic, maybe we should look only in the main connex component? Otherwise the user is stuck.

Reply:
I agree that the automatic slack node calculation should be in the main cc only

Reply:
This would require to change the architecture idea, since slack node and main connex component are determined in the same time. Feasible but not in the spirit of the beginning.

Besides, isn't that weird that the slack node in the whole network is not in the main connex component to begin with ? Are we sure we don't want to notify the user of that ?

Reply:
for the first point, then we should adapt the architecture if required.
For the second point, no it is not weird, we can have 2 sub network working in parallel in a simulation, it's just that dynawo is not able to handle that today.

Reply:
After discussion with @cossartque, this is what we'll do:
- if slack node is given in the IIDM file but not in the main connex component, send error
- if slack node is computed automatically, compute it first on the whole network then, if not in the main component, computes it only on the main component. That way, we avoid going through nodes of the network twice in a lot of cases but only in a worst case

Reply:
why not using the result of the connex component research during the slack node research?

Reply:
They are done on the same time. Each algorithm is computed node by node, meaning that we don't know the main connex component when we start the algorithm of slack node

Reply:
since everything is done on the main connex component, we might need later to first find it and then call algorithms only on it (and more later on all the connex components), but for today let's do this